docs/content/en/docs/patterns-and-best-practices/_index.md

```diff
@@ -120,3 +120,25 @@ might be a permission issue for some resources in another namespace.
 The `stopOnInformerErrorDuringStartup` has implication on [cache sync timeout](https://github.com/java-operator-sdk/java-operator-sdk/blob/114c4312c32b34688811df8dd7cea275878c9e73/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ConfigurationService.java#L177-L179)
 behavior. If true operator will stop on cache sync timeout. if `false`, after the timeout the controller will start
 reconcile resources even if one or more event source caches did not sync yet.  
+
+## Graceful Shutdown
+
+We can provide sufficient time for the reconciler to process and complete the currently ongoing events before shutting down. 
+There are two ways to support the graceful shutdown feature.
+
+The first method is to directly provide a timeout value to the `stop` method.
+
+```java
+final var operator = new Operator();
+
+operator.stop(Duration.ofSeconds(5));
+```
+
+The second method is to specify the timeout value by overriding the `ConfigurationService`.
+
+```java
+final var overridden = new ConfigurationServiceOverrider(config)
+    .withReconciliationTerminationTimeout(Duration.ofSeconds(5));
+
+final var operator = new Operator(overridden);
+```
```

Comment:
I believe it is beneficial for users to be aware of this feature, so I have documented it. 
If you think this documentation is unnecessary, please let me know and I will remove it.